example/src/main/java/io/netty/example/http2/tiles/Http2RequestHandler.java

```diff
@@ -0,0 +1,120 @@
+    private static final String IMAGE_COORDINATE_X = "x";
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) throws Exception {
+        QueryStringDecoder queryString = new QueryStringDecoder(request.uri());
+        String streamId = streamId(request);
+        int latency = toInt(firstValue(queryString, LATENCY), 0);
+        if (latency > MAX_LATENCY) {
+            sendBadRequest(ctx, streamId);
+            return;
+        }
+        String x = firstValue(queryString, IMAGE_COORDINATE_X);
+        String y = firstValue(queryString, IMAGE_COORDINATE_Y);
+        if (x == null || y == null) {
+            handlePage(ctx, streamId, latency);
+        } else {
+            handleImage(x, y, ctx, streamId, latency);
+        }
+    }
+
+    private void sendBadRequest(ChannelHandlerContext ctx, String streamId) {
+        FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, BAD_REQUEST, EMPTY_BUFFER);
+        streamId(response, streamId);
+        ctx.writeAndFlush(response);
+    }
+
+    private void handleImage(String x, String y, ChannelHandlerContext ctx, String streamId, int latency) {
+        ByteBuf image = ImageCache.INSTANCE.image(valueOf(x), valueOf(y));
+        FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, OK, image);
+        response.headers().set(CONTENT_TYPE, "image/jpeg");
+        sendResponse(ctx, streamId, latency, response);
+    }
+
+    private void handlePage(ChannelHandlerContext ctx, String streamId, int latency) {
+        byte[] body = Html.body(latency);
+        ByteBuf content = ctx.alloc().buffer(Html.HEADER.length + body.length + Html.FOOTER.length);
+        content.writeBytes(Html.HEADER);
+        content.writeBytes(body);
+        content.writeBytes(Html.FOOTER);
+        FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, OK, content);
+        response.headers().set(CONTENT_TYPE, "text/html; charset=UTF-8");
+        sendResponse(ctx, streamId, latency, response);
+    }
+
+    protected void sendResponse(final ChannelHandlerContext ctx, String streamId, int latency,
+            final FullHttpResponse response) {
+        setContentLength(response, response.content().readableBytes());
+        streamId(response, streamId);
+        ctx.executor().schedule(new Runnable() {
+            @Override
+            public void run() {
+                ctx.writeAndFlush(response);
+            }
+        }, latency, TimeUnit.MILLISECONDS);
+    }
+
+    private String streamId(FullHttpRequest request) {
+        return request.headers().get(HttpUtil.ExtensionHeaderNames.STREAM_ID.text());
+    }
+
+    private void streamId(FullHttpResponse response, String streamId) {
+        response.headers().set(HttpUtil.ExtensionHeaderNames.STREAM_ID.text(), streamId);
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+        System.err.print(cause);
```

Comment:
Is this consistent with other examples and exception handling?  Are we doing `e.printStackTrace()`?